glass/core/algorithm.py

```diff
@@ -93,3 +86,162 @@ def nnls(
         x[p] = sp
         x[~p] = 0
     return x
+
+
+def cov_clip(
+    cov: npt.NDArray[np.float64],
```

Comment:
Following #507, can we change this (and others) to adapt to it?
```suggestion
    cov: NDArray[np.float64],
```